tests/metal/nested-struct-fragment-output.slang

```diff
@@ -0,0 +1,40 @@
+//TEST:SIMPLE(filecheck=METAL): -target metal -stage fragment -entry fragmentMain
+//TEST:SIMPLE(filecheck=METALLIB): -target metallib -stage fragment -entry fragmentMain
+
+//METAL-DAG: color(0)
+//METAL-DAG: color(1)
+//METAL-DAG: color(2)
+//METALLIB: @fragmentMain
+
+RWStructuredBuffer<float> outputBuffer;
+
+struct NestedFragment
```

Comment:
When writing tests, 3 is the magical number. Test with 3-layer nested structs instead of just two.